src/Libraries/Microsoft.Extensions.AI.Abstractions/Functions/AIFunction.cs

```diff
@@ -1,20 +1,95 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Text.Json;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Shared.Collections;
+using Microsoft.Shared.Diagnostics;
 
 namespace Microsoft.Extensions.AI;
 
 /// <summary>Represents a function that can be described to an AI service and invoked.</summary>
 [DebuggerDisplay("{DebuggerDisplay,nq}")]
 public abstract class AIFunction : AITool
 {
-    /// <summary>Gets metadata describing the function.</summary>
-    public abstract AIFunctionMetadata Metadata { get; }
+    /// <summary>The description of the function.</summary>
+    private readonly string _description = string.Empty;
+
+    /// <summary>The JSON schema describing the function and its input parameters.</summary>
+    private readonly JsonElement _jsonSchema = AIJsonUtilities.DefaultJsonSchema;
+
+    /// <summary>Optional additional properties in addition to the named properties already available on this class.</summary>
+    private readonly IReadOnlyDictionary<string, object?> _additionalProperties = EmptyReadOnlyDictionary<string, object?>.Instance;
+
+    /// <summary>Gets the name of the function.</summary>
+    public abstract string Name { get; }
```

Comment:
@stephentoub I've now flattened the properties inside `AIFunction`. Using a protected constructor created akwardness in a number of implementations so I opted for making `Name` abstract instead. I've also marked all other properties easier for easier extension.